upgrade/steps.go

```diff
@@ -307,9 +303,11 @@ func UpgradeProviderVersion(
 func InformGitHub(
 	ctx context.Context, target *UpstreamUpgradeTarget, repo ProviderRepo,
 	goMod *GoMod, targetBridgeVersion, targetPfVersion Ref, tfSDKUpgrade string,
-) step.Step {
-	pushBranch := step.Cmd("git", "push", "--set-upstream",
-		"origin", repo.workingBranch).In(&repo.root)
+	osArgs []string,
+) error {
+	ctx = stepv2.WithEnv(ctx, &stepv2.Cwd{To: repo.root})
```

Comment:
I don't love this syntax for setting the working directory. At minimum, it would be nice to have a utility function, but honestly, I'd rather somehow wrap the function(s) that's executing with a CWD, to make it really obvious where it applies.

Reply:
What do you think about this?
```go
pwd = stepv2.Cmd(ctx, "pwd") // pwd ends with /base
stepv2.WithCwd(ctx, "./dir, func(ctx context.Context) {
     pwd = stepv2.Cmd(ctx, "pwd") // pwd ends with base/dir
})
pwd = stepv2.Cmd(ctx, "pwd") // pwd ends with /base again
```